# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContext.cpp
# Commit: f1fc2382346d
# Full Hash: f1fc2382346d163fe18a8fe5d323aafa759e1317
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2019-08-15 09:46:49
# Regressor Bug: 1562292
# File Overlap Count: 1
# Description:
#   Bug 1562292: Part 1b.1 - Fix FindWithSpecialName with inaccessible "_parent". r=nika
#   
#   The special name "_parent" resolves to the parent window when one exists, or
#   the current target window if it does not. Either way, it returns null if the
#   target is inaccessible.
# ==============================================================================

diff -r d513683e1fdb -r f1fc2382346d docshell/base/BrowsingContext.cpp
--- a/docshell/base/BrowsingContext.cpp	Tue Jul 30 17:30:55 2019 -0700
+++ b/docshell/base/BrowsingContext.cpp	Thu Aug 01 18:45:47 2019 -0700
@@ -550,9 +550,10 @@
   }
 
   if (aName.LowerCaseEqualsLiteral("_parent")) {
-    return mParent && aRequestingContext.CanAccess(mParent.get())
-               ? mParent.get()
-               : this;
+    if (mParent) {
+      return aRequestingContext.CanAccess(mParent) ? mParent.get() : nullptr;
+    }
+    return this;
   }
 
   if (aName.LowerCaseEqualsLiteral("_top")) {
